datadog_lambda/tracing.py

```diff
@@ -415,7 +465,10 @@ def is_legacy_lambda_step_function(event):
         return False
 
     event = event.get("Payload")
-    return "Execution" in event and "StateMachine" in event and "State" in event
+    return isinstance(event, dict) and (
```

Comment:
Updates the legacy lambda check to account for `_datadog` case and added the extra check for safety

Reply:
If a customer use Legacy Lambda and legacy schema (meaning not using v1), they won't have `_datadog`.